src/drive/web/modules/views/OnlyOffice/useCreateFile.jsx

```diff
@@ -9,7 +9,7 @@ import {
   makeMimeByClass
 } from 'drive/web/modules/views/OnlyOffice/helpers'
 
-const useCreateFile = (folderId, fileClass) => {
+const useCreateFile = (folderId, fileClass, { enable } = { enable: true }) => {
```

Comment:
Did you just add an arg to the hook because you can't make a condition into the component that call this hook? 

You should refactor / split the component, not adding complexity to the hook

Reply:
Now, I'm using the `onEnter` property of the route to check permissions rather than conditionally calling the hook. If the user doesn't have write permissions, his redirect to root folder with the paywall